script/src/verify.rs

```diff
@@ -455,6 +467,267 @@ impl<'a, DL: CellDataProvider + HeaderProvider> TransactionScriptsVerifier<'a, D
         Ok(cycles)
     }
 
+    fn make_snapshot(
+        &self,
+        mut vm: Machine<'a>,
+        group: &ScriptGroup,
+        current: (ScriptGroupType, Byte32),
+        remain: Vec<(ScriptGroupType, Byte32)>,
+        current_cycles: Cycle,
+        limit_cycles: Cycle,
+    ) -> Result<TransactionSnapshot, Error> {
+        let snap = make_snapshot(&mut vm.machine).map_err::<Error, _>(|e| {
```

Comment:
Communicate with the vm team, if there is a problem here, it should crash directly, it must be a bug